graylog2-server/src/main/java/org/graylog2/cluster/NodeServiceImpl.java

```diff
@@ -51,29 +54,25 @@ public Node.Type type() {
     }
 
     @Override
-    public String registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String hostname) {
-        Map<String, Object> fields = Maps.newHashMap();
-        fields.put("last_seen", Tools.getUTCTimestamp());
-        fields.put("node_id", nodeId);
-        fields.put("type", type().toString());
-        fields.put("is_leader", isLeader);
-        fields.put("transport_address", httpPublishUri.toString());
-        fields.put("hostname", hostname);
-
-        try {
-            Node node;
-            try {
-                // Update existing node object.
-                final String objectId = byNodeId(nodeId).getId();
-                node = new NodeImpl(new ObjectId(objectId), fields);
-            } catch (NodeNotFoundException e) {
-                // Create new node object.
-                node = new NodeImpl(fields);
-            }
-            return save(node);
-        } catch (ValidationException e) {
-            throw new RuntimeException("Validation failed.", e);
-        }
+    public boolean registerServer(String nodeId, boolean isLeader, URI httpPublishUri, String hostname) {
+        final Map<String, Object> fields = Map.of(
+                "$set", Map.of(
+                        "node_id", nodeId,
+                        "type", type().toString(),
+                        "is_leader", isLeader,
+                        "transport_address", httpPublishUri.toString(),
+                        "hostname", hostname
+                ),
+                "$currentDate", lastSeenField
+        );
```

Comment:
I'd consider using` com.mongodb.client.model.Updates`, namely its set() and currentTimestamp() methods to make the code cleaner, get rid of special mongo-magic strings and `lastSeenField` constant.